### DISCUSSION TITLE
Is it possible to discover the top K motifs in a single time series data without specifying window size ?

### DISCUSSION THREAD

**Discussion by tnkjm** [TDAmeritrade/stumpy#587] [Q&A] [answered]
Hello, 

My time series data has the following characteristics
- The X coordinate of the wrist is recorded every 0.03 seconds when the worker repeats the same task multiple times.
- Because the workers are human, the number of seconds of each task varies slightly from one to another (so I think I cannot specify window size)

I am trying to detect a single pattern from this data. The assumption is that we do not know how many times the work is repeated by the worker.

I became interested in stumpy because of the mention of Find Top-K Motifs in the [tutorial](https://stumpy.readthedocs.io/en/latest/Tutorial_STUMPY_Basics.html#Find-Top-K-Motifs).

I've tried `stumpy.motifs`, but I realize I need to specify the window size to compute `P – Matrix Profile of T` to be entered into stumpy.motifs.
https://stumpy.readthedocs.io/en/latest/api.html#motifs

Should I try another method?

Thank you.

**Comment by seanlaw** [comment, marked as answer]
@tnkjm Thank you for your question and welcome to the STUMPY community. To address your question, the short answer is "no" but note that the window size does not have to be exact (i.e., you just need to choose an approximate window size that is slightly larger than your pattern). If the task takes roughly the same amount of time across multiple users (with some small variance) then, assuming that the wrist coordinate is roughly similar, you should  still be able to capture this with a large enough window size. Trying to capture a task that can span multiple time scales is a really, really hard problem. [Here is a paper](https://www.mdpi.com/2571-9394/4/1/13/htm) that I recently came across (not written by me) that claims to address this problem (i.e., when there is variance in your window size) but I have not tried it myself.

**Reply by tnkjm** [reply]
Thank you for your response and for sharing a paper.
